### PR TITLE
msvc: Update vcpkg manifest

### PR DESCRIPTION
**Pull request by hebasto**
This PR updates the vcpkg manifest baseline from the [2023.08.09 Release ](https://github.com/microsoft/vcpkg/releases/tag/2023.08.09) to the [2024.09.30 Release](https://github.com/microsoft/vcpkg/releases/tag/2024.09.30), with the following package changes:
- `boost`: 1.82.0#2 --> 1.85.0#1,2
- `qt5`: 5.15.10#5 -> 5.15.15
- `sqlite3`: 3.42.0#1 --> 3.46.1
- `zeromq`: 2023-06-20#1 --> 4.3.5#2

The previous update was made in https://github.com/bitcoin/bitcoin/pull/28938.

For additional minor improvements, please refer to the commit messages.

### CODE REVIEW

**Comment by hebasto** [comment]
Friendly ping @sipsorcery @hodlinator ;)

**Review by hodlinator — COMMENTED** [review]
Tested on Windows 11 Home:
```
> rmdir /S build
> cmake -B build --preset vs2022-static
> cmake --build build --config Release -j<X>
> build\src\qt\Release\bitcoin-qt.exe (started successfully)
```

**Review by Marcolaris — APPROVED** [review]

File: vcpkg.json [hunk 1]
```diff
@@ -1,46 +1,64 @@
 {
   "$schema": "https://raw.githubusercontent.com/microsoft/vcpkg-tool/main/docs/vcpkg.schema.json",
-  "builtin-baseline": "9edb1b8e590cc086563301d735cae4b6e732d2d2",
-  "overrides":[
-    {"name": "libevent", "version": "2.1.12#7"},
-    {"name": "liblzma", "version": "5.4.1#1"}
-  ],
+  "$comment": "The builtin-baseline corresponds to 2024.09.30 Release",
```
[the remark this answers is not in the document]
Comment (hebasto) [reply vcpkg.json:3]: This `"$comment"` field actually comments the following line:https://github.com/bitcoin/bitcoin/blob/fad685ce54d39a333a6c75641b5d519cb0345959/vcpkg.json#L3-L4

[the remark this answers is not in the document]
Comment (hodlinator) [reply vcpkg.json:3]: Saw that, but it mentions it by name so I thought maybe keeping the lexicographical order the first commit tries to introduce would have precedent.

File: vcpkg.json
[the remark this answers is not in the document]
Comment (hebasto) [reply vcpkg.json:1]: I've dropped mentioning of lexicographical order from the commit message. The goal of the reordering is to minimize diffs for invocations like `vcpkg add port libqrencode`.

**Review by Skate-Cloud — APPROVED** [review]

**Review by fanquake — APPROVED** [review]
ACK f6577b717416191e65ba8221be57f1a48d74e5d7

**Pull request merged**